sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -1467,10 +1494,12 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
   }
 
   test("SPARK-36980: Insert support query with CTE") {
-    withTable("t") {
-      sql("CREATE TABLE t(i int, part1 int, part2 int) using parquet")
-      sql("INSERT INTO t WITH v1(c1) as (values (1)) select 1, 2, 3 from v1")
-      checkAnswer(spark.table("t"), Row(1, 2, 3))
+    withSQLConf(SQLConf.USE_NULLS_FOR_MISSING_DEFAULT_COLUMN_VALUES.key -> "false") {
```

Comment:
Same as above, could we also add a version of this test with the conf set to "true" to show that the intended behavior works? (This should be the same as the test code in this case before this PR.)